// @flatfile/react/dist/dev.index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.mount = void 0;
var react_1 = __importDefault(require("react"));
var react_dom_1 = require("react-dom");
var index_1 = require("./index");
var config = {
    type: 'test import',
    fields: [
        { label: 'name', key: 'name' },
        { label: 'Email', key: 'email' },
    ],
};
var license = 'aa921983-4db2-4da1-a580-fbca0b1c75b2';
var customer = { userId: '12345' };
var data = [
    {
        data: {
            firstName: 'foo',
            lastName: 'bar',
            email: 'john@doe.com',
            address: 'US',
        },
        errors: [{ key: 'firstName', message: 'foo' }],
    },
    {
        data: {
            firstName: 'Alice',
            lastName: 'bar',
            email: 'alice@doe.com',
            address: 'US',
        },
        errors: [{ key: 'lastName', message: 'another message' }],
    },
    {
        data: {
            firstName: 'Henry',
            lastName: 'Person',
            email: 'henry@example.com',
            address: 'US',
        },
        errors: [{ key: 'email', message: 'example.com is not allowed' }],
    },
];
exports.mount = function (doc) {
    var container = doc.createElement('div');
    doc.body.appendChild(container);
    var errorState = true;
    react_dom_1.render(react_1.default.createElement(index_1.FlatfileButton, { settings: config, licenseKey: license, customer: customer, onData: function (results) { return __awaiter(void 0, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, new Promise(function (resolve, reject) {
                        setTimeout(function () {
                            if (errorState) {
                                reject('rejected');
                                errorState = false;
                            }
                            else {
                                resolve();
                            }
                        }, 3000);
                    })];
            });
        }); }, onRecordChange: function (record) {
            var _a;
            return { name: { value: (_a = record.name) === null || _a === void 0 ? void 0 : _a.toString().toUpperCase() } };
        }, onRecordInit: function (record) {
            var _a;
            return { name: { value: (_a = record.name) === null || _a === void 0 ? void 0 : _a.toString().substring(0, 2) } };
        }, fieldHooks: {
            email: function (values) {
                return values.map(function (_a) {
                    var item = _a[0], index = _a[1];
                    return [{ value: item + '@' }, index];
                });
            },
        }, onCancel: function () {
            console.log('cancel');
        }, render: function (importer, launch) {
            return react_1.default.createElement("a", { onClick: launch }, "upload here");
        }, source: data }, "Start FF"), container);
};
exports.mount(document);
